docs/examples/gluster-s3-storage-template/README.md

```diff
@@ -2,111 +2,136 @@
+
+ Replace `NAMESPACE` and `ADMIN_KEY` parameters with your configuration.
+Here, `NAMESPACE` is the project and `ADMIN_KEY` is used for authorization to access Heketi service.
 
 ```
-oc create -f ./gluster-s3-storageclass.yaml
+oc create secret generic heketi-${NAMESPACE}-admin-secret --from-literal=key=${ADMIN_KEY} --type=kubernetes.io/glusterfs
 ```
 
-Available at 
-[gluster-s3-storageclass.yaml](./gluster-s3-storageclass.yaml)
+As an optional step, the Secret can be labelled. This is useful to be able to select the secret as part of a general query like `oc get
+--selector=glusterfs` and allows the secret to be removed programatically by the `gk-deploy` tool.
 
-## Start glusters3 service using template
+Replace `NAMESPACE` parameter with your configuration.
 
 ```
-oc new-app gluster-s3-template.yaml  --param=S3_ACCOUNT=testvolume  --param=S3_USER=adminuser --param=S3_PASSWORD=itsmine --param=VOLUME_CAPACITY=2Gi
+oc label --overwrite secret heketi-${NAMESPACE}-admin-secret glusterfs=s3-heketi-${NAMESPACE}-admin-secret gluster-s3=heketi-${NAMESPACE}-admin-secret
 ```
 
-Note: adjust parameters according to your needs.
+Create a GlusterFS StorageClass as below, replacing the `HEKETI_URL` and `NAMESPACE` parameters with your configuration.
+Also, provide `STORAGE_CLASS` to give a storage class name.
+`HEKETI_URL` is the REST url to access GlusterFs cluster.
+`NAMESPACE` is the project.
+`STORAGE_CLASS` is the StorageClass name provided by admin.
 
+```
+sed  -e 's/${HEKETI_URL}/heketi-store-project1.cloudapps.mystorage.com/g'  -e 's/${STORAGE_CLASS}/gluster-s3-store/g'      -e  's/${NAMESPACE}/store-project1/g'   deploy/ocp-templates/gluster-s3-storageclass.yaml | oc create -f -
+```
 
-If you wish to make use of a GlusterFS StorageClass other than `s3storageclass`, add another parameter of the form:
+Available at
+[gluster-s3-storageclass.yaml](../../../deploy/ocp-templates/gluster-s3-storageclass.yaml)
 
+#### Create backend PVCs
+
+Now, create PVCs using the StorageClass. Replace `STORAGE_CLASS` with the above created one and adjust `VOLUME_CAPACITY` as per your needs.
 
 ```
---param=STORAGE_CLASS=<your storage class name>
+sed -e 's/${VOLUME_CAPACITY}/2Gi/g'  -e  's/${STORAGE_CLASS}/gluster-s3-store/g'  deploy/ocp-templates/gluster-s3-pvcs.yaml | oc create -f -
+persistentvolumeclaim "gluster-s3-claim" created
+persistentvolumeclaim "gluster-s3-meta-claim" created
 ```
 
+Available at
+[gluster-s3-pvcs.yaml](../../../deploy/ocp-templates/gluster-s3-pvcs.yaml)
 
+### 2. Start glusters3 service
 
-Available at:
-[gluster-s3-template.yaml](./gluster-s3-template.yaml)
+Launch S3 storage service.
+Set S3_ACCOUNT name, S3_USER name, S3_PASSWORD according to the user wish. S3_ACCOUNT is the S3 account which will be created and associated with GlusterFS volume. S3_USER is the user created to access the above account and S3_PASSWORD is for Authorization of the S3 user.  
+PVC and META_PVC are persistentvolumeclaim(s) obtained from above step.
```

Comment:
Combine this line with the previous line, and enclose all VARIABLE names with backticks (`` ` ``).

Reply:
Done